src/Workspaces/Core/Portable/FlowAnalysis/LValueFlowCaptureProvider.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.FlowAnalysis
+{
+    /// <summary>
+    /// Helper class to detect <see cref="IFlowCaptureOperation"/>s that are l-value captures.
+    /// </summary>
```

Comment:
can you give an example?

Reply:
Done.

```
    /// <summary>
    /// Helper class to detect <see cref="IFlowCaptureOperation"/>s that are l-value captures.
    /// L-value captures are essentially captures of a symbol's location/address.
    /// Corresponding <see cref="IFlowCaptureReferenceOperation"/>s which share the same
    /// <see cref="CaptureId"/> as this flow capture, dereferences and writes to this locaion
    /// subsequently in the flow graph.
    /// For example, consider the below code:
    ///     a[i] = x ?? a[j];
    /// The control flow graph contains an initial flow capture of "a[i]" to capture the l-value
    /// of this array element:
    ///     FC0 (a[i])
    /// Then it evaluates the right hand side, which can have different
    /// values on different control flow paths, and the resultant value is then written
    /// to the captured location:
    ///     FCR0 = result
    /// </summary>
    /// <remarks>
    /// NOTE: This type is a temporary workaround for [https://github.com/dotnet/roslyn/issues/31007](https://github.com/dotnet/roslyn/issues/31007)
    /// and it can be deleted once that feature is implemented.
    /// </remarks>
```

---
In reply to: [230606751](https://github.com/dotnet/roslyn/pull/30777#discussion_r230606751) [](ancestors = 230606751)